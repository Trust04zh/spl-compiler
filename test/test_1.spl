int var_1;

int fun(char arg_1) {
    if (arg_1 == 'c') {
        return 1;
    } else {
        return 0;
    }
}

int mian() {
    float var_2;
    char var_3;
    struct S_1 {
        int attr_1, attr_2;
    } s[2];
    s[0].attr_1 = 3;
    s[0].attr_2 = 4;
    var_1 = s[1].attr_1 + s[1].attr_2;
    s[1].attr_1 = var_1;
    var_3 = 'c';
    s[1].attr_2 = var_1 + fun(var_3);
    return 1;
}
