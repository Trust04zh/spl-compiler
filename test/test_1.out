Error type 21 at Line 4: only int variables can do boolean operations
